test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/V3DetailControlModelTests.cs

```diff
@@ -254,18 +345,41 @@ public V3PackageSolutionDetailControlModelTests(GlobalServiceProvider sp, V3Pack
 #pragma warning disable ISB001 // Dispose of proxies
             serviceBroker.Setup(x => x.GetProxyAsync<INuGetProjectManagerService>(It.Is<ServiceJsonRpcDescriptor>(d => d.Moniker == NuGetServices.ProjectManagerService.Moniker), It.IsAny<ServiceActivationOptions>(), It.IsAny<CancellationToken>()))
                 .ReturnsAsync(projectManagerService.Object);
+            serviceBroker.Setup(
+                x => x.GetProxyAsync<INuGetSearchService>(
+                    NuGetServices.SearchService,
+                    It.IsAny<ServiceActivationOptions>(),
+                    It.IsAny<CancellationToken>()))
+                .Returns(new ValueTask<INuGetSearchService>(mockSearchService.Object));
 #pragma warning restore ISB001 // Dispose of proxies
 
+            ServiceLocator.InitializePackageServiceProvider(this);
```

Comment:
This will cause flakey errors if any other test also calls `ServiceLocator.InitializePackageServiceProvider`, and the tests run in parallel.

Can we pass the `IAsyncServiceProvider` instance to classes via their class constructor, instead of using the static `ServiceLocator` class?

Reply:
This sounds like a DCR? I see we have code that references the static `ServiceLocator` already, so even if I change the test pattern we're using, it won't mitigate the problem you raised - right?

Example, https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetSolutionManagerService.cs#L68

Reply:
Removed all `ServiceLocator` references and the tests still pass, so either this is no longer necessary or was never necessary.